src/Microsoft.VisualStudio.Web.CodeGeneration.Utils/Workspaces/RoslynWorkspace.cs

```diff
@@ -42,25 +42,24 @@ public class RoslynWorkspace : Workspace
             //    AddProjectReference(file, configuration);
             //}
 
-            AddMetadataReferences(projectDependencyProvider, id);
+            AddMetadataReferences(context, id);
         }
 
-        private ProjectId AddProject(MsBuildProjectFile projectFile, string configuration)
+        private ProjectId AddProject(IProjectContext context, string configuration)
         {
-            var fullPath = projectFile.FullPath;
+            var fullPath = context.ProjectFullPath;
 
-            var projectName = Path.GetFileNameWithoutExtension(fullPath);
             var projectInfo = PInfo.Create(
                 ProjectId.CreateNewId(),
                 VersionStamp.Create(),
-                projectName,
-                projectName,
+                context.ProjectName,
+                context.ProjectName,
```

Comment:
Assembly name may differ from project name. You can get assembly file name from `Path.GetFileNameWithoutExtensions(context.AssemblyFullPath)`. maybe `IProjectContext.AssemblyName` would be useful?